### DISCUSSION TITLE
这个系统的主函数在哪呀

### DISCUSSION THREAD

**Discussion by TerrenceYi** [alibaba/GraphScope#3110] [Q&A] [answered]
我是一名图计算研0新生，本科不是读的计算机，想问问这个sssp.h运行的主函数在哪个代码呀，导师要求我分析具体图算法运行时的通信模型，希望好心人指条明路

**Comment by yecol** [comment]
Hi @TerrenceYi, 
GraphScope provides a Python interface. Python triggers analytical functions.
e.g., you may find a pagerank invocation in the ReadMe. 

If you want to dive into the code, 
sssp starts here: 
https://github.com/alibaba/GraphScope/blob/main/python/graphscope/analytical/app/sssp.py

it will run sssp algorithms defined in libgrape-lite, over the assigned graph.

**Comment by yecol** [comment, marked as answer]
If you want to learn more about the communication mechanism, 
the repo libgrape-lite is a good start. 
main: https://github.com/alibaba/libgrape-lite/blob/master/examples/analytical_apps/run_app.cc
sssp: https://github.com/alibaba/libgrape-lite/blob/master/examples/analytical_apps/sssp/sssp.h
worker with communication: https://github.com/alibaba/libgrape-lite/blob/master/grape/worker/worker.h

**Comment by TerrenceYi** [comment]
Thank you very much for your generous help
